src/app/api/api_v1/endpoints/utils/verifications.py

```diff
@@ -35,3 +37,19 @@ def verify_project_robot_type_template_assignment(project_obj: Project) -> bool:
         if robot_type.generation_template_id is None:
             return False
     return True
+
+
+def verify_template_version_increase_on_content_change(generation_template_obj: GenerationTemplate,
+                                                       generation_template_in: GenerationTemplateUpdate) -> bool:
+    """
+    Verifies that on template update the content can only be changed when the version was increased as well
+    :param generation_template_obj: Current template in the database
+    :param generation_template_in:  Update object of the template to update
+    :return: True, if template update ensures the constraint
+    """
+    if generation_template_in.content is not None and generation_template_obj.content != generation_template_in.content:
+        if generation_template_in.version is not None \
+                and generation_template_in.version > generation_template_obj.version:
+            return True
+        return False
+    return True
```

Comment:
```suggestion
    if generation_template_in.content is None or generation_template_obj.content == generation_template_in.content:
        return True
    if generation_template_in.version is None or generation_template_in.version <= generation_template_obj.version:
        return False
    return True
```
to reduce cyclomatic complexity